src/OutputReaders/field_time_series_indexing.jl

```diff
@@ -170,6 +177,18 @@ end
                              from_loc, from_grid, times, backend, time_indexing)
 
     to_time = to_time_index.time
+    ñ, n₁, n₂ = interpolating_time_indices(time_indexing, times, to_time)
+    interp = InterpolationIndices(ñ, n₁, n₂, length(times))
```

Comment:
why do we need this inside the kernel?

Reply:
This does not change the kernel code. What we have done is reorganize the code so that we always use `InterpolationIndices`. However, there is a second way to interpolate which does not require using `interpolating_time_indices`, and instead constructing `InterpolationIndices` directly -- for example, outside the kernel.

Reply:
In other words with this new feature, this way of writing the code is _required_ to eliminate code duplication. Otherwise, we have to write `interpolate` twice.